frontend/src/components/AbstractModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dialog, DialogContent, DialogTitle, DialogActions, Typography, makeStyles, IconButton } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import * as React from 'react';
+import Button from './Button';
+
+
+export type modalType =  {
+    title: string;
+    description: string; 
+    confirmText: string;
+    cancelText: string;
+    isOpen: boolean;
+    handleConfirm: (e) => void;
+    handleCancel: (e) => void;
+};
+
+const useStyles = makeStyles(theme =>({
+    modal:{
+        position:'absolute',
+        top: theme.spacing(5),
+        padding: theme.spacing(2)
+    },
+    modalContent:{
+        textAlign: 'center'
+    },
+    modalAction:{
+        justifyContent: 'center'
+    },
+    modalTitle: {
+        position: 'absolute',
+        top: theme.spacing(2),
+        right: theme.spacing(0.5)
+    }
+}))
+
+
+export default function AbstractModal(props) {
+    const [modal, setModal] = React.useState<modalType>(props);
```

Comment:
```suggestion
    const [modal, setModal] = React.useState<ModalType>(props);
```

Also why are all of the props part of the state? It appears that the only property that can change is `isOpen`